SOC/common/simulation.py

```diff
@@ -84,23 +93,44 @@ def AvalancheLoop(self) -> dict:
         AvalancheSize = self.visited.sum()
         return dict(AvalancheSize=AvalancheSize, number_of_iterations=number_of_iterations)
 
-    def run(self, N_iterations: int) -> dict:
+    def run(self, N_iterations: int, filename: str  = None) -> dict:
         """
         Simulation loop. Drives the simulation, possibly starts avalanches, gathers data.
 
         :param N_iterations:
         :type N_iterations: int
         :rtype: dict
-        """
+        :param filename: filename for saving snapshots. if None, saves to memory; by default if False, makes something like array_Manna_2019-12-17T19:40:00.546426.zarr
+        :type filename: str
+        """
+        if filename is False:
```

Comment:
My zawsze zapisujemy plik?

Reply:
Zapomniałem poprawić docstringa; default jest `None`, czyli idzie in memory. Trafna uwaga.